.buildkite/scripts/sbt_dependency_tree.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+# -*- encoding: utf-8
+
+import json
+import os
+
+
+class Project:
+    def __init__(self, name, folder, dependencies):
+        self.name = name
+        self.folder = folder
+        self.dependencies = dependencies
+
+    def __repr__(self):
+        return "<%s name=%r folder=%r dependencies=%r>" % (
+            type(self).__name__,
+            self.name,
+            self.folder,
+            ",".join(sorted(d.name for d in self.dependencies)),
+        )
+
+    def all_folders(self):
+        folders = set([self.folder])
+        for d in self.dependencies:
+            folders |= d.all_folders()
+        return folders
+
+    def depends_on(self, other):
+        if self == other:
+            return True
+        if other in self.dependencies:
+            return True
+        for d in self.dependencies:
+            return d.depends_on(other)
+        return False
+
+
+class Repository:
+    def __init__(self, metadata_dir):
+        self.metadata_dir = metadata_dir
+        self.projects = {}
+
+        self._get_all_projects()
+
+    def _get_all_projects(self):
+        for f in os.listdir(self.metadata_dir):
+            self.get_project(name=os.path.splitext(f)[0])
+
+    def get_project(self, name):
```

Comment:
bikeshed: `get_project` indicates to me this returns something, i would prefer this to be called `init_project` or something like that